packages/storage/amplify_storage_s3_dart/lib/src/amplify_storage_s3_dart_impl.dart

```diff
@@ -94,8 +85,7 @@ class AmplifyStorageS3Dart extends StoragePluginInterface
       );
     }
 
-    _prefixResolver ??= StorageAccessLevelAwarePrefixResolver(
-      delimiter: _delimiter,
+    _pathResolver = S3PathResolver(
```

Comment:
todo: remove. `_pathResolver` is set immediately below this. This is duplicated code.